File: core/src/com/alta_v2/facade/tiledMapApi/TiledMapCoreApi.java
package com.alta_v2.facade.tiledMapApi;

import com.alta_v2.physics.task.MovementDirection;
import com.alta_v2.physics.task.resultObserver.TaskResultObserver;

public interface TiledMapCoreApi {

    /**
     * Gets the ID of object which is target of player.
     */
    Integer getPlayerPurpose();

    /**
     * Tries to perform movement of player.
     *
     * @param direction - the direction of movement to be perform.
     */
    TaskResultObserver performPlayerMovement(MovementDirection direction);

    /**
     * Tries to perform movement of NPC with given ID.
     *
     * @param npcId     - the id of NPC to be moved.
     * @param direction - the direction of movement to be perform.
     */
    TaskResultObserver performNpcMovement(int npcId, MovementDirection direction);

}

File: core/src/com/alta_v2/facade/tiledMapApi/TiledMapCoreApi.kt
package com.alta_v2.facade.tiledMapApi

import com.alta_v2.physics.task.MovementDirection
import com.alta_v2.physics.task.resultObserver.TaskResult

interface TiledMapCoreApi {
    /**
     * Gets the ID of object which is target of player.
     */
    val playerPurpose: Int?

    /**
     * Tries to perform movement of player.
     *
     * @param direction - the direction of movement to be perform.
     */
    fun performPlayerMovement(direction: MovementDirection): TaskResult?

    /**
     * Tries to perform movement of NPC with given ID.
     *
     * @param npcId     - the id of NPC to be moved.
     * @param direction - the direction of movement to be perform.
     */
    fun performNpcMovement(npcId: Int, direction: MovementDirection): TaskResult?

    /**
     * Performs focusing of NPC with given id to player.
     */
    fun performFocusNpcOnPlayer(npcId: Int): TaskResult?
}